src/Sylius/Behat/Context/Api/Admin/ManagingExchangeRatesContext.php

```diff
@@ -0,0 +1,374 @@
+    {
+        $this->client->buildUpdateRequest('exchange_rates', $exchangeRate->getId());
+    }
+
+    /**
+     * @Given I am browsing exchange rates of the store
+     * @When I browse exchange rates
+     * @When I browse exchange rates of the store
+     */
+    public function iBrowseExchangeRatesOfTheStore(): void
+    {
+        $this->client->index('exchange_rates');
+    }
+
+    /**
+     * @When /^I specify its ratio as (-?[0-9\.]+)$/
+     * @When I don't specify its ratio
+     */
+    public function iSpecifyItsRatioAs(?float $ratio = null): void
+    {
+        if($ratio !== null) {
+            $this->client->addRequestData('ratio', $ratio);
+        }
+    }
+
+    /**
+     * @When I choose :currencyCode as the source currency
+     */
+    public function iChooseAsTheSourceCurrency(string $currencyCode): void
+    {
+        $this->client->addRequestData('sourceCurrency', '/new-api/currencies/' . $currencyCode);
+    }
+
+    /**
+     * @When I choose :currencyCode as the target currency
+     */
+    public function iChooseAsTheTargetCurrency(string $currencyCode): void
+    {
+        $this->client->addRequestData('targetCurrency', '/new-api/currencies/' . $currencyCode);
+    }
+
+    /**
+     * @When I (try to) add it
+     */
+    public function iAddIt(): void
+    {
+        $this->client->create();
+    }
+
+    /**
+     * @When I change ratio to :ratio
+     */
+    public function iChangeRatioTo(string $ratio): void
+    {
+        $this->client->addRequestData('ratio', $ratio);
+    }
+
+    /**
+     * @When I save my changes
+     */
+    public function iSaveMyChanges(): void
+    {
+        $this->client->update();
+    }
+
+    /**
+     * @When I delete the exchange rate between :sourceCurrency and :targetCurrency
+     */
+    public function iDeleteTheExchangeRateBetweenAnd(CurrencyInterface $sourceCurrency, CurrencyInterface $targetCurrency) : void
+    {
+        /** @var ExchangeRateInterface */
+        $exchangeRate = $this->getExchangeRateBetweenCurrencies($sourceCurrency->getCode(), $targetCurrency->getCode());
+
+        $this->client->delete('exchange_rates', $exchangeRate->getId());
+    }
+
+    /**
+     * @When I choose :currency as a currency filter
+     */
+    public function iChooseCurrencyAsACurrencyFilter(CurrencyInterface $currency): void
+    {
+        $exchangeRates = [];
+        $this->client->filter('exchange_rates', 'sourceCurrency.code', $currency->getCode());
+        $exchangeRates = array_merge($exchangeRates, $this->client->getCollection());
+        $this->client->filter('exchange_rates', 'targetCurrency.code', $currency->getCode());
+        $exchangeRates = array_merge($exchangeRates, $this->client->getCollection());
+
+        $this->sharedStorage->set('filtered_exchange_rates', $exchangeRates);
+    }
+
+    /**
+     * @When I filter
+     */
+    public function iFilter(): void
+    {
+        //Intentionally left blank to fulfill context expectation
```

Comment:
I'm not sure about this comment, in name of this step you have ' I  am doing something' but in the body of function you have 'i'm doing nothing' :)